opentracing-jdbc/opentracing-jdbc/src/test/java/org/zalando/opentracing/jdbc/DataSourceTracerTest.java

```diff
@@ -0,0 +1,236 @@
+             final Statement statement = connection.createStatement()) {
+
+            statement.addBatch("INSERT INTO users(id, name) VALUES (1, 'Alice')");
+            statement.addBatch("INSERT INTO users(id, name) VALUES (2, 'Bob')");
+            statement.executeBatch();
+        }
+
+        final List<MockSpan> spans = tracer.finishedSpans();
+        final MockSpan span = getOnlyElement(spans);
+
+        assertThat(span.operationName(), is("executeBatch"));
+        assertThat(span.tags(), hasEntry("db.statement",
+                "INSERT INTO users(id, name) VALUES (1, 'Alice')\nINSERT INTO users(id, name) VALUES (2, 'Bob')"));
+    }
+
+    @Test
+    void shouldTracePreparedBatch() throws SQLException {
+        try (final Connection connection = original.getConnection();
+             final Statement statement = connection.createStatement()) {
+
+            statement.execute("CREATE TABLE users(id BIGINT, name TEXT)");
+        }
+
+        final DataSource dataSource = unit.trace(original);
+
+        final String sql = "INSERT INTO users(id, name) VALUES (?, ?)";
+        try (final Connection connection = dataSource.getConnection();
+             final PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setInt(1, 1);
+            statement.setString(2, "Alice");
+            statement.addBatch();
+
+            statement.setInt(1, 2);
+            statement.setString(2, "Bob");
+            statement.addBatch();
+
+            statement.executeBatch();
+        }
+
+        final List<MockSpan> spans = tracer.finishedSpans();
+        final MockSpan span = getOnlyElement(spans);
+
+        assertThat(span.operationName(), is("executeBatch"));
+        assertThat(span.tags(), hasEntry("db.statement", "INSERT INTO users(id, name) VALUES (?, ?)"));
+    }
+
+    @Test
+    void shouldUseCustomOperationName() throws SQLException {
+        try (final Connection connection = original.getConnection()) {
+            final String sql = "CREATE ALIAS MATRIX FOR \"org.zalando.opentracing.jdbc.Matrix.create\";";
+            try (final PreparedStatement statement = connection.prepareStatement(sql)) {
+                statement.execute();
+            }
+        }
+
+        final DataSource dataSource = unit
+                .withOperationName(new StoredProcedureOperationName())
+                .trace(original);
+
+        try (final Connection connection = dataSource.getConnection()) {
+            try (final Statement statement = connection.createStatement()) {
+                statement.execute("SELECT * FROM MATRIX(4)");
+                statement.execute("SELECT 1");
+            }
+        }
+
+        final List<MockSpan> spans = tracer.finishedSpans();
+        assertThat(spans, hasSize(2));
+
+        assertThat(spans.get(0).operationName(), is("MATRIX"));
+        assertThat(spans.get(1).operationName(), is("execute"));
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/whiskeysierra/tracer/pullRequest?prid=4563925)